app/controllers/amp_articles_controller.rb

```diff
@@ -0,0 +1,47 @@
+class AmpArticlesController < ActionController::Base
+  def show
+    if (@article = Article.find_by_permalink(params[:from])).present?
+      if @article.supports_amp || params[:no_redirect]
+        respond_to do |format|
+          format.html do
+            render 'articles/amp/show'
+          end
+        end
+      else
+        redirect_to article_url(@article.permalink)
+      end
+    else
+      render 'errors/404', status: 404
+    end
+  end
+
+  def article_body(article)
+    body = article.html(:body)
+    page = Nokogiri::HTML.fragment(body)
+    page = process_img_tag(page)
+    page = process_iframe_tag(page)
+    page
+  end
+  helper_method :article_body
```

Comment:
I originally was cool with this being in the controller given, well, it's publify, but you're right to request this to be put somewhere else.

My other issue though is that Nokogiri's api for modifying content doesn't lend well to building into a PORO. Every change is done by modifying the receiver, there's no scope for modelling the object on something domain related. So your class is really just being used to wrap some code that's more procedural in style.